### ISSUE TITLE
(Possible) problem in provenance_redesign migration

### ISSUE THREAD

**Issue by giovannipizzi** [aiidateam/aiida_core#2406]
I found this log message after performing some migrations - it would be useful to understand if this is expected or a bug in the migration, and fix it:
```
# 2019-01-21T07:41:28.242953
# Violation reason: found calculation nodes with a type string that could not be mapped onto a known entry point
# Performed action: inferred `process_type` for all calculation nodes, using fallback for unknown entry points

UUID                                  type (old)                       process_type (fallback)
------------------------------------  -------------------------------  -------------------------
6aa6c01a-e8b6-4933-a8fb-e967823bbbce  calculation.job.JobCalculation.  .JobCalculation
```

**Comment by sphuber** [comment]
This does not look like a bug in the migration as such, just it is a case that we did not consider. Apparently at some point it was possible to store an instance of the base `JobCalculation` class. To "fix" this, we should add an additional update query in the provenance redesign that finds those entries and transforms them into `CalcJobNode` entries instead as that would be the current analogue

**Comment by giovannipizzi** [comment]
Ok, let's then update the migration - I'll keep this issue in the project.

**Comment by sphuber** [comment]
@giovannipizzi  Taking a closer at this, this may not actually be a bug in the migration of the data itself but neither in the process type inference step before the migration, which is where this message comes from. The type string `calculation.job.JobCalculation.` indicates that the plugin class was `JobCalculation`, which clearly cannot be mapped onto a known entry point. So the message is correct in that sense. The only thing that we should decide is whether the fallback value is correct. The logic assumes that it contains at least a module followed by a class, i.e. `calculation.job.somemodule.SomeClass.`, the fallback process type of which will be `somemodule.SomeClass`. Since in this case there is no module but just a class, the fallback starts with a `.`. Given that production databases should not have `calculation.job.JobCalculation.` node types, should we change the logic?

**Comment by sphuber** [comment]
Fixed in PR #2428
